src/components/Post/index.jsx

```diff
@@ -189,6 +189,8 @@ function Post(prop) {
       }}
     >
       <ErrorBoundary>
+        {
+          !profile &&  
```

Comment:
we don't need to hide the header completely, we just had to hide the visit profile option, as you had mentioned there as well. Also, instead of prop drilling, you can instead check the current route as well, uselocation... or there might be other better ways as well